go-packages/meep-model/model.go

```diff
@@ -916,10 +1072,65 @@ func (m *Model) UpdateScenario() {
 	}
 }
 
+// Node Type validation functions
+
+func isScenario(typ string) bool {
+	return typ == NodeTypeScenario
+}
+
+func isDomain(typ string) bool {
+	return typ == NodeTypeOperator || typ == NodeTypeOperatorCell
+}
+
 func isDefaultZone(typ string) bool {
 	return typ == "COMMON"
 }
 
+func isZone(typ string) bool {
+	return typ == NodeTypeZone
+}
+
 func isDefaultNetLoc(typ string) bool {
 	return typ == "DEFAULT"
 }
+
+func isNetLoc(typ string) bool {
+	return typ == NodeTypePoa || typ == NodeTypePoa4G || typ == NodeTypePoa5G || typ == NodeTypePoaWifi
+}
+
+func isPhyLoc(typ string) bool {
+	return typ == NodeTypeCloud || typ == NodeTypeEdge || typ == NodeTypeFog || typ == NodeTypeUE
+}
+
+func isProc(typ string) bool {
+	return typ == NodeTypeCloudApp || typ == NodeTypeEdgeApp || typ == NodeTypeUEApp
+}
+
+func validateParentType(nodeType string, parentType string) bool {
```

Comment:
I get all the function call in if-else format... but i wonder if exploding the nodeType in a switch case is not cleaner... since in the isPhyLoc and isProc, there is still a need to check for specific nodeTypes

Reply:
For maintainability I prefer to stick with the function calls